execution/block-partitioner/src/v2/state.rs

```diff
@@ -0,0 +1,349 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    pre_partition::start_txn_idxs,
+    v2::{
+        conflicting_txn_tracker::ConflictingTxnTracker,
+        counters::MISC_TIMERS_SECONDS,
+        types::{PreParedTxnIdx, SenderIdx, ShardedTxnIndexV2, StorageKeyIdx, SubBlockIdx},
+    },
+    Sender,
+};
+use aptos_types::{
+    block_executor::partitioner::{
+        CrossShardDependencies, RoundId, ShardId, ShardedTxnIndex, SubBlock,
+        TransactionWithDependencies, TxnIndex,
+    },
+    state_store::state_key::StateKey,
+    transaction::analyzed_transaction::{AnalyzedTransaction, StorageLocation},
+};
+use dashmap::DashMap;
+use rayon::{
+    iter::{IntoParallelIterator, ParallelIterator},
+    ThreadPool,
+};
+use std::{
+    collections::HashSet,
+    mem,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, Mutex, RwLock,
+    },
+};
+
+/// All the parameters, indexes, temporary states needed in a `PartitionerV2` session wrapped in a single struct
+/// to make async drop easy.
+pub struct PartitionState {
+    // Params/utils from partitioner.
+    pub(crate) num_executor_shards: ShardId,
+    pub(crate) num_rounds_limit: usize,
+    pub(crate) avoid_pct: u64,
+    pub(crate) merge_discarded: bool,
+    pub(crate) thread_pool: Arc<ThreadPool>,
+
+    // Holding all the txns.
+    pub(crate) txns: Vec<RwLock<Option<AnalyzedTransaction>>>,
+
+    // Pre-partitioning results.
+    pub(crate) pre_partitioned: Vec<Vec<PreParedTxnIdx>>,
+    pub(crate) start_txn_idxs_by_shard: Vec<PreParedTxnIdx>,
```

Comment:
There is value. Otherwise
```
    pub(crate) start_index_matrix: Vec<Vec<PreParedTxnIdx>>,
    pub(crate) new_txn_idxs: Vec<RwLock<TxnIndex>>,
```
becomes
```
    pub(crate) start_index_matrix: Vec<Vec<TxnIdx>>,
    pub(crate) new_txn_idxs: Vec<RwLock<TxnIndex>>,
```
which is terrible...

Reply:
As discussed offline, let's rename this to `PrePartitionedTxnIdx`